### PR TITLE
ascii-draw: init at 0.3.0

### PR DESCRIPTION
**Pull request by Aleksanaa**
## Description of changes

Not so bad

image

```
    ┌╶╶╶╶╶╶╶╶╶╶╶╶╶╶╶╶╶╶╶╶╶╶╶╶╶╶╶╶╶╶╶╶╶╶┐
    ╎                                  ╎
    ╎                                  ╎
    ╎    _      _ ___  _ ____  ____    ╎
    ╎   / \  /|/ \\  \///  _ \/ ___\   ╎
    ╎   | |\ ||| | \  / | / \||    \   ╎
    ╎   | | \||| | /  \ | \_/|\___ |   ╎
    ╎   \_/  \|\_//__/\\\____/\____/   ╎
    ╎                                  ╎
    ╎                                  ╎
    ╎                                  ╎
    └╶╶╶╶╶╶╶╶╶╶╶╶╶╶╶╶╶╶╶╶╶╶╶╶╶╶╶╶╶╶╶╶╶╶┘
```
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/as/ascii-draw/package.nix [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    makeWrapperArgs+=("''${gappsWrapperArgs[@]}")
+  '';
+
+  meta = {
+    description = "An app to draw diagrams or anything using only ASCII";
+    homepage = "https://github.com/Nokse22/ascii-draw";
+    license = lib.licenses.gpl3Plus;
+    mainProgram = "ascii-draw";
+    maintainers = with lib.maintainers; [ aleksana ];
+    platforms = lib.platforms.unix;
```
Comment (wegank) [inline pkgs/by-name/as/ascii-draw/package.nix:64]:
```suggestion
    platforms = lib.platforms.linux;
```

https://www.gnu.org/software/gnulib/manual/html_node/bindtextdomain.html

Comment (Aleksanaa) [reply pkgs/by-name/as/ascii-draw/package.nix:64]: oops

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 300045` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ascii-draw</li>
  </ul>
</details>

**Comment by wegank** [comment]
Doesn't work on my machine:

```bash
$ LANG=C ~/.cache/nixpkgs-review/pr-300045/results/ascii-draw/bin/ascii-draw
(process:1835145): Gtk-WARNING **: 23:26:11.568: Theme parser error: gtk.css:5:1-128: Failed to import: Error opening file /nix/store/a9m8ms83b4kgv711v200py09vljj5jmq-gnome-themes-extra-3.28/share/themes/Adwaita/gtk-4.0/gtk.css: No such file or directory
Traceback (most recent call last):
  File "/nix/store/h3hn2vn76ilxbgimqv6xvh7p7cvmpq5a-ascii-draw-0.3.0/share/ascii-draw/ascii_draw/main.py", line 156, in do_activate
    self.win = AsciiDrawWindow(application=self)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/h3hn2vn76ilxbgimqv6xvh7p7cvmpq5a-ascii-draw-0.3.0/share/ascii-draw/ascii_draw/window.py", line 267, in __init__
    os.makedirs(directory_path, exist_ok=True)
  File "<frozen os>", line 225, in makedirs
PermissionError: [Errno 13] Permission denied: '/palettes'
```

File: pkgs/by-name/as/ascii-draw/package.nix [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    rev = "v${version}";
+    hash = "sha256-vI+j8OuQ3b6La0+7wWeoUtBal24dazlN/T0Bng5TgMo=";
+  };
+
+  # Silently fail autosaving to flatpak directory
+  # Autosaving won't work, but program runs normally otherwise
+  # https://github.com/Nokse22/ascii-draw/issues/31
+  postPatch = ''
+    substituteInPlace src/window.py \
+      --replace-fail "/var/data" ""
```
Comment (wegank) [inline pkgs/by-name/as/ascii-draw/package.nix:30]: Looks like this turned `/var/data/palettes` into `/palettes`?

Comment (Aleksanaa) [reply pkgs/by-name/as/ascii-draw/package.nix:30]: I'd admit I'm dumb

```
=> ls /
bin  boot  dev  etc  home  lib  lib64  nix  palettes  proc  root  run  srv  sys  tmp  usr  var
```

**Comment by Aleksanaa** [comment]
Forget it, I'll try to solve this problem correctly.

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 300045` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ascii-draw</li>
  </ul>
</details>

**Review by wegank — APPROVED** [review]
Is the patch upstreamable?

**Comment by Aleksanaa** [comment]
I'm pretty sure it won't work inside flatpak, but I'm not currently interested in testing flatpak's behavior with these directories. At the same time, my implementation is also relatively sloppy. Let's first see how upstream developers respond to my issue.

**Pull request merged**